File: Php/app/Urls/BaseClasses/PlayerUrlModel.php
<?php
namespace app\Urls\BaseClasses{

	use entities\Player;
	use app\UrlFormatter;
	use entities\Homegame;

	class PlayerUrlModel extends UrlModel{

		public function __construct($format, Homegame $homegame, Player $player){
			$url = UrlFormatter::formatPlayer($format, $homegame, $player);
			parent::__construct($url);
		}

	}

}

File: Web/Models/Url/PlayerUrlModel.cs
using Core.Classes;

namespace Web.Models.Url{

	class PlayerUrlModel : UrlModel{

		public PlayerUrlModel(string format, Homegame homegame, Player player){
			var url = UrlFormatter.FormatPlayer(format, homegame, player);
			SetUrl(url);
		}

	}

}